lib/src/license_detection/license.dart

```diff
@@ -0,0 +1,117 @@
+import 'dart:collection';
+import 'dart:io';
+
+import 'package:meta/meta.dart';
+import 'package:pana/src/license_detection/tokenizer.dart';
+
+@sealed
+class License {
+  /// Name of the license, is empty in case of unknown license.
+  final String licenseName;
+
+  /// Original text from the license file.
+  final String content;
+
+  /// Normalized [Token]s created from the original text.
+  final List<Token> tokens;
+
+  /// A map of tokens and their count.
+  final Map<String, int> occurences;
+
+  License._(this.content, this.tokens, this.occurences, this.licenseName);
+
+  factory License.parse(String licenseName, String content) {
+    final tokens = tokenize(content);
+    final table = generateFrequencyTable(tokens);
+    return License._(content, tokens, table, licenseName);
+  }
+}
+
+// Commented out as they are not yet needed in this stage of detection.
+// class Checksum {
+//   /// Text for which the hash value was generated.
+//   final String text;
+
+//   /// [Crc-32][1] checksum value generated for text.
+//   ///
+//   /// [1]: https://en.wikipedia.org/wiki/Cyclic_redundancy_check
+//   final int crc32;
+
+//   Checksum(this.text, this.crc32);
+// }
+
+class Result {
+  /// Name of the license detected from unknown text.
+  final String name;
+
+  /// Tokens that were found in the unknown text that matched to tokens
+  /// in any of the corpus license.
+  final List<Token> tokens;
+
+  /// Confidence score of the detected license.
+  final double confidenceScore;
+
+  Result(this.name, this.tokens, this.confidenceScore);
+}
+
+/// Genearates a frequency table for the give list of tokens.
+Map<String, int> generateFrequencyTable(List<Token> tokens) {
+  var map = HashMap<String, int>();
+
+  for (var token in tokens) {
+    if (map.containsKey(token.value)) {
+      map[token.value] = map[token.value]! + 1;
+    } else {
+      map[token.value] = 1;
+    }
```

Comment:
Just a small thing... either solution works fine :smile: 

```suggestion
   map[token.value] = map.putIfAbsent(token.value, () => 0) + 1;
```

Personally, I'm a huge fan of `Map.putIfAbsent`, see: https://api.dart.dev/stable/2.13.4/dart-core/Map/putIfAbsent.html

Reply:
I'm also part of your little fan club from now on 😄.